### PR TITLE
[FIX] (website_)crm: fix compute on default and duplicate without mmr

### PR DESCRIPTION
**Pull request by jeh-odoo**
# Purpose
- Fix the traceback raised when creating a lead because of the
visitor_page_count method when self.ids is empty on default compute.
- Fix the traceback when duplicating a lead when the MMR option
is not activated.
- Fix traceback when trying to merge leads.

See underlying commits for details.

task-2325629

### CODE REVIEW

File: addons/website_crm/models/crm_lead.py [hunk 1]
```diff
@@ -12,17 +12,20 @@ class Lead(models.Model):
...
-                    FROM crm_lead l
-                    JOIN crm_lead_website_visitor_rel lv ON l.id = lv.crm_lead_id
-                    JOIN website_visitor v ON v.id = lv.website_visitor_id
-                    JOIN website_track p ON p.visitor_id = v.id
-                    WHERE l.id in %s
-                    GROUP BY l.id"""
-        self.env.cr.execute(sql, (tuple(self.ids),))
-        page_data = self.env.cr.dictfetchall()
-        mapped_data = {data['lead_id']: data['page_view_count'] for data in page_data}
+        if self.ids:
```
Comment (awa-odoo) [inline addons/website_crm/models/crm_lead.py:15]: A bit shorter:
```
mapped_data = {}
if self.ids:
    ...
```

File: addons/crm/models/crm_lead.py [hunk 2]
```diff
@@ -635,7 +635,7 @@ def copy(self, default=None):
         # Do not assign to an archived user
         if not self.user_id.active:
             default['user_id'] = False
-        return super(Lead, self.with_context(context)).copy(default=default)
+        return super(Lead, self.with_context(context).sudo()).copy(default=default)
```
Comment (awa-odoo) [inline addons/crm/models/crm_lead.py:638]: I really don't see the link between the commit message and the fix.
Adding a random sudo() seems suspicious.

**Review by awa-odoo — COMMENTED** [review]
Hello @jeh-odoo 

Here are a few comments.
Also, seems like you're missing one of the commits in the PR description.

Thanks for your work!

**Review by awa-odoo — COMMENTED** [review]
Here is a commit message suggestion.
I already set it to green.

Maybe also add to your PR description after the bullet points:
> See underlying commits for details.

Thanks!

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo retry

**Pull request closed without merging**